### DISCUSSION TITLE
Displaying modification dates for markdown pages?

### DISCUSSION THREAD

**Discussion by StefanSalewski** [thuliteio/doks#1242] [General]
Generally, I like it very much when web pages display a creation date, or date of last editing. For some pages that might not be needed, but often it is useful. I have wasted already much time studying pages found by Google, just to discover later that the page was written years ago and the content is fully outdated.

For HTML pages it is easy to add a date display somewhere, I even found a way to use a Hugo shortcode to add the current date automatically.

But most of my content files are markdown. OK, I could insert a date somewhere in the text. But I would prefer a dedicated location, and if possible use of the date value from frontmatter lastmod entry. I spent some time searching for something like that, but could not find it. One possible solution might be the footer menu, which show typically an entry "Privacy Policy". But then the date field would be select-able, which is generally not desired. (Well, when thinking about it, a select-able date field, containing a link to diff of latest changes would be an option.) But I would prefer a plain read-only date field, like the   "Brought to you by" text at the lower right of each page.

**Comment by h-enk** [comment]
When you set: 
- in `config/_default/hugo.toml` (line 8): `enableGitInfo = true`, and
- in  `config/_default/params.toml`  (line 75): `lastMod = true` 

The [last modification date](https://gohugo.io/methods/page/lastmod/) of a page is shown  — see for example: https://hub.botstudioo.com/docs/functional-system/nodes/

Note, you can override the current behavior by copying `node_modules/@hyas/doks-core/layouts/partials/main/last-modified.html` to `layouts/partials/main/last-modified.html` and make your changes in the latter.

**Comment by StefanSalewski** [comment]
Thank you very much for the very fast and helpful reply. That seems to be exactly the desired behavior.

**Comment by StefanSalewski** [comment]
Actually, I prefer not to use Git for my tiny pages. My actual text is only a few kB in size. Just using Git increases already the directory size a lot, and hosting the full page content at GitHub seems to be wasting a lot of GitHub resources.

By modifying  layouts/partials/main/last-modified.html I found a good solution for myself, see below.

Generally, I think the following behavior would be great:

If enableGitInfo = true, use and display dates from Git.
If enableGitInfo = false, use .Date and .Lastmod from frontmatter.
If frontmatter does not define the values, do not show them.

Additionally, an option to show dates at the top would be nice.

With Google and some GPT-4 support I found the following solution, which is exactly what I want:

```
salewski@hx90 ~/nimprogramming.bak $ cat layouts/partials/main/last-modified.html
{{ if or .Date .Lastmod }}
  <div class="last-modified">
    <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-calendar"><rect x="3" y="4" width="18" height="18" rx="2" ry="2"></rect><line x1="16" y1="2" x2="16" y2="6"></line><line x1="8" y1="2" x2="8" y2="6"></line><line x1="3" y1="10" x2="21" y2="10"></line></svg>
    {{ if .Date }}
      Created: {{ .Date | time.Format ":date_medium" }}
    {{ end }}
    {{ if and .Lastmod (or (not .Date) (ne .Lastmod .Date)) }}
      Last edited: {{ .Lastmod | time.Format ":date_medium" }}
    {{ end }}
  </div>
{{ end }}
```

I was really surprised how well GPT-4 understands this syntax.

**Reply by h-enk** [reply]
Thanks for sharing! Yes, I agree there's room for improvement in the logic for showing the last modification date — I will record that in an issue (to be taken up),

**Reply by h-enk** [reply]
https://github.com/gethyas/doks/issues/1244

**Reply by Backspaze** [reply]
> Additionally, an option to show dates at the top would be nice.
> 
> With Google and some GPT-4 support I found the following solution, which is exactly what I want:
> 
> ```
> salewski@hx90 ~/nimprogramming.bak $ cat layouts/partials/main/last-modified.html
> {{ if or .Date .Lastmod }}
>   <div class="last-modified">
>     <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-calendar"><rect x="3" y="4" width="18" height="18" rx="2" ry="2"></rect><line x1="16" y1="2" x2="16" y2="6"></line><line x1="8" y1="2" x2="8" y2="6"></line><line x1="3" y1="10" x2="21" y2="10"></line></svg>
>     {{ if .Date }}
>       Created: {{ .Date | time.Format ":date_medium" }}
>     {{ end }}
>     {{ if and .Lastmod (or (not .Date) (ne .Lastmod .Date)) }}
>       Last edited: {{ .Lastmod | time.Format ":date_medium" }}
>     {{ end }}
>   </div>
> {{ end }}
> ```

I was looking for a solution for this and found this thread. I took your solution and modified it a bit, in part to address the part about showing the dates at the top, which is something that I also wanted. The result is shown in the screenshot below, and further below is how I got there.

<img width="272" height="191" alt="bild" src="https://github.com/user-attachments/assets/557c9012-c0f0-4213-a8a2-f32085086f39" />

- I copied `node_modules\@thulite\doks-core\layouts\single.html` to `layouts\single.html` and modified it as below, to add the dates to the top and remove them from the bottom.
```html
...
<h1>{{ .Title }}</h1>
	{{ if site.Params.doks.lastMod -}}
		{{ partial "main/last-modified.html" . }}
	{{ end -}}			
...
<div class="page-footer-meta d-flex flex-column flex-md-row justify-content-between">
	<!--{{ if site.Params.doks.lastMod -}}
		{{ partial "main/last-modified.html" . }}
	{{ end -}}-->
...
````
- I copied `node_modules\@thulite\doks-core\layouts\_partials\main\last-modified.html` to `layouts\_partials\main\last-modified.html` and modified the contents as below.
```html
{{ if or .Date .Lastmod }}
  <div class="last-modified">
    <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-calendar"><rect x="3" y="4" width="18" height="18" rx="2" ry="2"></rect><line x1="16" y1="2" x2="16" y2="6"></line><line x1="8" y1="2" x2="8" y2="6"></line><line x1="3" y1="10" x2="21" y2="10"></line></svg>
    {{ if .Date }}
      {{ i18n "published" }}: {{ .Date | time.Format ":date_medium" }}
    {{ end }}
	<br>
    <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-refresh-cw"><polyline points="23 4 23 10 17 10"></polyline><polyline points="1 20 1 14 7 14"></polyline><path d="M3.51 9a9 9 0 0 1 14.85-3.36L23 10M1 14l4.64 4.36A9 9 0 0 0 20.49 15"></path></svg>
    {{ if and .Lastmod (or (not .Date) (ne .Lastmod .Date)) }}
      {{ i18n "updated" }}: {{ .Lastmod | time.Format ":date_medium" }}
    {{ end }}
  </div>
{{ end }}
```
- I edited `\assets\scss\common\_custom.scss` as below.
```scss
.last-modified {
    margin-top: 1rem;
    margin-bottom: 1rem;
}
```

**Reply by h-enk** [reply]
Thanks for sharing!
